app/classifier/drawing-tools/root.cjsx

```diff
@@ -15,6 +16,9 @@ SEMI_MODAL_UNDERLAY_STYLE =
 module.exports = React.createClass
   displayName: 'DrawingToolRoot'
 
+  contextTypes:
+    store: React.PropTypes.object
```

Comment:
Do we have to use `context` here? Have you checked if `store` is available in the tree?

Reply:
`react-redux` uses `context` to pass the store into connected components, but I'm happy to remove it here if you can suggest a better way to pass it into the modal form.

Reply:
Doesn't the `connect` API for `react-redux`do that?

Reply:
Yeah, a connected component gets the store from `this.context` inside `connect()`.

Reply:
`TaskTranslations` is the connected component here, by the way.

Reply:
If `TaskTranslations` is `connect`ed it has `store` already, you don't need to pass it to it. Sorry, I'm probably missing something.

Reply:
Try removing `store={@context.store}` from `TaskTranslations` and see what happens.

Reply:
My best guess here is that the `StickyModalForm` component doesn't pass `context` to its children.

Reply:
If you remove context, you should see the following error in the console when you try a drawing task that pops up a modal form: `Uncaught Error: Could not find "store" in either the context or props of "Connect(TaskTranslations)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(TaskTranslations)".`

Reply:
It's probably me being slow, sorry, but if `TaskTranslations` is `connect`ed it has `store` already, you don't need to pass it to it.

Reply:
<img width="1082" alt="screen shot 2018-01-11 at 10 25 02" src="https://user-images.githubusercontent.com/8989606/34820933-fa8cf7f0-f6b9-11e7-96a3-fc85198696d2.png">
The react dev tools show the same properties, whether `store` is passed or not. So I'm not sure what's doing there. As far as I can tell the functionality is not affected by removing `store` and, when I draw I don't get any error, just a `warning.js?8a56:36 Warning: Unknown prop `tool` on <g> tag. Remove this prop from the element`.

Reply:
Are you trying a drawing task that has subtasks, like the cat face tool in I Fancy Cats or the circle tool in the dev classifier?

Reply:
Your screenshot is debugging the drawing task, not the subtasks.

Reply:
Ah I see what you mean now. This is really annoying: `connect` should do the job and we definitely shouldn't need `contextTypes`, but, as you said, something in between is not playing ball.

Reply:
`Connect` gets the store (and subscription) from `context.store`, then looks for `props.store` if context isn't available.
https://github.com/reactjs/react-redux/blob/e170b2c54fbb796a32c087dcb9a1325abe5a370c/src/components/connectAdvanced.js#L81-L87

Dialogs like `StickyModalForm` don't pass context down to their children, which is why `Connect` throws an error here.

Reply:
Do you mean all dialogs do that in general? Why? Can you point to a link please?

Reply:
Yes, because they create elements outside of the current tree. Easiest way to see this is to open a tutorial on a zooniverse project, then inspect the popup with the React dev tools and look at the `Connect` component.

Reply:
You should see that `context` is an empty object. If you inspect a `Connect` component elsewhere on a zooniverse project page, you'll see that `context.store` is defined.

Reply:
I think he's talking about the dialogs from the modal-form (https://github.com/zooniverse/modal-form) library. We could open an issue in that repo to add context to what is passed down. It should be simple enough to add to the Base class.

Reply:
React 16 introduces [Portals](https://reactjs.org/docs/portals.html) to work with children outside of the current parent node.

Reply:
Oh yeah, I forgot about that. I'd really love it if we could get to React 16. One thing at a time though.

Reply:
Gotcha, thanks. Here's the issue https://github.com/zooniverse/modal-form/issues/32

Reply:
This affects all modals, not just the modal form library. For example, tutorials, mini-courses and field guides also have to be passed the store explicitly as a prop.

Reply:
It depends on the design of the modal. I've seen some React modal libraries that are designed to be a child in the tree and use CSS to position it. If it's written to add the React element outside of the tree, then yes. That's how the modal-form is written as well as the dialog.cjsx version in PFE (which Brian used as a base for modal-form I think).

Reply:
Yeah. Field guides are using `react-pullout` which also has the same problem. I thought tutorials were using PFE's built-in Dialog component but maybe not. I remember raising with Brian that modal-form was a bad fit here, because tutorials aren't forms and shouldn't use `<form>` elements. That was on the list of things to fix, but I don't remember if we ever got to it.

Reply:
There's a [`PassContext`](/zooniverse/Panoptes-Front-End/blob/master/app/components/pass-context.cjsx) component which takes the PFE context and passes it to its children. Would using that be a better solution?